flags.go

```diff
@@ -68,14 +54,6 @@ func defineFlagVar() *FlagVar { //nolint:funlen
 		"The resources.limits.cpu for skr webhook.")
 	flag.StringVar(&flagVar.virtualServiceName, "virtual-svc-name", "kcp-events",
 		"Name of the virtual service resource to be reconciled by the watcher control loop.")
-	flag.Var(newNamespacedNameVar(&flagVar.gatewayNamespacedName), "gateway-ns-name",
-		"Namespaced name of the gateway resource that the virtual service will use. "+
-			"Format: <namespace>/<name>. Example: \"my-namespace/my-gateway\".")
-	flag.Var(newLabelSelectorVar(&flagVar.gatewaySelector).
-		withDefault(operatorv1alpha1.DefaultIstioGatewaySelector()),
-		"gateway-selector", "Label selector of the gateway resource that the virtual service will use. "+
-			"Ignored if gateway-ns-name flag is specified. Format: K8s label selector expression. "+
-			"Example: \"label1=value1,label2=value2\"")
```

Comment:
I think the system now should ensure we have a distinct VirtualService for different Watcher CRs?
If so, I'd rather remove the flags in exact opposite way: Gateway selectors can be defined statically with CLI flags, because we don't need distinct gateways for different Watcher CRs.
But the VirtualServices must be different, so defining their names with a single static flag doesn't make sense.

Reply:
Regarding VirtualService, you are right, current design not make sense to just use a single VirtualService, but as I mentioned in the related issues https://github.com/kyma-project/lifecycle-manager/issues/326, we will refactoring it to dedicated VirtualService in this followup issue https://github.com/kyma-project/lifecycle-manager/issues/336, but not here.

Regarding Gateway, if we only consider Gateway for our components (KLM and KMM), to configure it as a flag could make sense, but KLM also have to take care of other components' Watcher, for those components, we are not limiting the gateway setup, it's up to the teams to define their contract, what KLM needs to do is just build the connection between gateway and VirtualService, so use Gateway selector which defined in Watcher CR is more generic approach, that's the reason remove those two flags.